xwiki-rendering-macros/xwiki-rendering-macro-ctsreport/src/main/java/org/xwiki/rendering/internal/macro/ctsreport/Test.java

```diff
@@ -65,15 +65,12 @@ public String toString()
     @Override
     public boolean equals(Object object)
     {
-        if (object == null) {
+        if (object == null || (object.getClass() != getClass())) {
```

Comment:
Since Test is not really designed to be extended, a slight improvement would be

    if (!(object instanceof Test))

(it cannot be an instance of Test if it's null)